### DISCUSSION TITLE
Time tracking

### DISCUSSION THREAD

**Discussion by dhindman-sst** [Budibase/budibase#8993] [Help] [answered]
I’m planning an app that incorporates user time into the app for reporting purposes. Ideally I’d be able to display the amount of time a user has spent logged into a specific app and also Budibase as a whole (if they have access to multiple apps, for example) over different periods of time. An example would be a user seeing that they’ve been in a certain app for x hours and y minutes. I’d also have a managerial screen listing all app users with their total times for the day, week and month. 

I haven’t noticed anything in the docs about user logs. How could I achieve the above? Is it possible to trigger a new time stamp database row upon actions like logging into an app, switching apps and logging out?

**Comment by melohagan** [comment, marked as answer]
Hey @dhindman-sst 

This is currently something that isn't available in Budibase, but it is on the roadmap for next year, hopefully over the next six months or so.

**Reply by dhindman-sst** [reply]
Thank you. I'm experimenting with using the [Countup](https://github.com/melohagan/budibase-component-countup) component in the meantime to at least approximate time on page to my users.
